packages/components/src/components/Calculator/asyncActions.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  SqValue,
+  SqCalculator,
+  SqError,
+  SqProject,
+  SqValuePath,
+  result,
+  Env,
+} from "@quri/squiggle-lang";
+
+import {
+  CalculatorState,
+  CalculatorAction,
+  calculatorReducer,
+  optionalResultValue,
+  allFieldResults,
+  allFieldValuesAreValid,
+  initialCalculatorState,
+} from "./calculatorReducer.js";
+
+import { ViewProviderDispatch } from "../SquiggleViewer/ViewerProvider.js";
+
+const runSquiggleCode = async (
+  code: string,
+  environment: Env
+): Promise<result<SqValue, SqError>> => {
+  const project = SqProject.create();
+  if (environment) {
+    project.setEnvironment(environment);
+  }
+  const sourceId = "calculator";
+  project.setSource(sourceId, code);
+  await project.run(sourceId);
+  const output = project.getOutput(sourceId);
+  if (output.ok) {
+    const result: result<SqValue, SqError> = {
+      ok: true,
+      value: output.value.result,
+    };
+    return result;
+  } else {
+    return output;
+  }
+};
+
+const updateFnValue = async (
```

Comment:
Doesn't have to be async, I don't see any async calls here.

Reply:
Good point, changed. it uses a call, ``calc.run``, that could take a while - but right now this isn't async, so it doesn't matter anyway.